### PR TITLE
feat(events): add event search

### PR DESCRIPTION
**Pull request by steebchen**

### CODE REVIEW

File: frontend/app/src/pages/main/events/index.tsx [hunk 1]
```diff
@@ -170,10 +172,6 @@ function EventsTable() {
   //   }
   // }, [listEventsQuery.data?.pagination]);
 
-  if (listEventsQuery.isLoading) {
-    return <Loading />;
-  }
```
Comment (steebchen) [inline frontend/app/src/pages/main/events/index.tsx:178]: removes global loading and instead moves it to the table

File: internal/repository/prisma/event.go [hunk 2]
```diff
@@ -50,6 +50,11 @@ func (r *eventRepository) ListEvents(tenantId string, opts *repository.ListEvent
 		TenantId: *pgTenantId,
 	}
 
+	if opts.Search != nil {
+		queryParams.Search.String = *opts.Search
```
Comment (abelanger5) [inline internal/repository/prisma/event.go]:
```suggestion
		queryParams.Search = sqlchelpers.TextFromStr(opts.Search)
```

^ can use this method which is meant to make this a little cleaner (setting `.Valid` isn't ideal) - just noticed that the `TextFromStr` method panics if the scan fails, which I'm not sure is ideal, but we also don't want to case on an error each time we do a simple string conversion, so perhaps it's alright?

File: internal/repository/prisma/event.go [hunk 3]
```diff
@@ -50,6 +50,11 @@ func (r *eventRepository) ListEvents(tenantId string, opts *repository.ListEvent
 		TenantId: *pgTenantId,
 	}
 
+	if opts.Search != nil {
+		queryParams.Search = sqlchelpers.TextFromStr(opts.Search)
+		queryParams.Search.Valid = true
```
Comment (steebchen) [inline internal/repository/prisma/event.go]:
```suggestion
```

Comment (steebchen) [reply internal/repository/prisma/event.go]: @abelanger5 this can be removed right :D

Comment (abelanger5) [reply internal/repository/prisma/event.go]: Yep!

File: frontend/app/src/components/molecules/data-table/data-table-toolbar.tsx [hunk 4]
```diff
@@ -22,26 +23,28 @@ interface DataTableToolbarProps<TData> {
...
     <div className="flex items-center justify-between">
       <div className="flex flex-1 items-center space-x-2">
-        {/* <Input
-          placeholder="Filter tasks..."
-          value={(table.getColumn("title")?.getFilterValue() as string) ?? ""}
-          onChange={(event) =>
-            table.getColumn("title")?.setFilterValue(event.target.value)
-          }
+        <Input
+          placeholder="Search for tasks..."
```
Comment (abelanger5) [inline frontend/app/src/components/molecules/data-table/data-table-toolbar.tsx]:
```suggestion
          placeholder="Search for events..."
```

**Review by abelanger5 — CHANGES_REQUESTED** [review]
Looks great - just one small language change 👍

**Review by abelanger5 — APPROVED** [review]

**Review by abelanger5 — APPROVED** [review]

**Pull request merged**